src/foundry/client/documents/abstract/document-collection.d.mts

```diff
@@ -206,38 +254,37 @@ declare namespace DocumentCollection {
 
   interface SearchOptions extends InexactPartial<_SearchOptions> {}
 
+  /** Used in the {@linkcode DocumentCollection.getSearchableFields} return type. */
+  // TODO: infer from schema all `StringField`s and subclasses that are `textSearch: true`
```

Comment:
Hmmm... file an issue I guess? I'm not sure I like the idea of having a registry of all fields for paranoia reasons though.